src/Microsoft.AspNetCore.Dispatcher/EndpointMiddleware.cs

```diff
@@ -18,8 +18,12 @@ public EndpointMiddleware(RequestDelegate next)
         public async Task Invoke(HttpContext context)
         {
             var feature = context.Features.Get<IDispatcherFeature>();
-            await feature.RequestDelegate(context);
-            await _next(context);
+            var requestDelegate = feature.RequestDelegate(context);
+            await requestDelegate;
+            if (requestDelegate == null)
+            {
+                await _next(context);
+            }
```

Comment:
This is still not right. Think about what would happen if `feature.RequestDelegate` is null